src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopyTestWrapper.java

```diff
@@ -39,27 +50,84 @@
 
     private String connectionString;
 
+    private ISQLServerDataSource dataSource;
+
     BulkCopyTestWrapper(String connectionString) {
         this.connectionString = connectionString;
     }
 
     /**
+     * Returns boolean value <code>true</code> if connection object is used for testing bulk copy
      * 
-     * @return
+     * @return isUsingConnection
      */
     public boolean isUsingConnection() {
         return isUsingConnection;
     }
 
     /**
+     * Returns boolean value <code>true</code> if XA connection object is used for testing bulk copy
+     * 
+     * @return isUsingXAConnection
+     */
+    public boolean isUsingXAConnection() {
+        return isUsingXAConnection;
+    }
+
+    /**
+     * Returns boolean value <code>true</code> if pooled connection object is used for testing bulk copy
      * 
+     * @return isUsingPooledConnection
+     */
+    public boolean isUsingPooledConnection() {
+        return isUsingPooledConnection;
+    }
+
+    /**
      * @param isUsingConnection
      *        <code>true</code> if connection object should be passed in BulkCopy constructor <code>false</code> if
      *        connection string is to be passed to constructor
      */
-    public void setUsingConnection(boolean isUsingConnection) {
+    public void setUsingConnection(boolean isUsingConnection, ISQLServerDataSource ds) {
         this.isUsingConnection = isUsingConnection;
         testName += "isUsingConnection=" + isUsingConnection + ";";
+        setDataSource(ds);
+    }
+
+    /**
+     * Sets boolean property if XA Connection must be used to test bulk Copy
+     * 
+     * @param isUsingXAConnection
+     *        <code>true</code> if XA connection object should be passed in BulkCopy constructor <code>false</code> if
+     *        otherwise
+     */
+    public void setUsingXAConnection(boolean isUsingXAConnection, ISQLServerDataSource ds) {
+        if (!isUsingConnection) {
+            // Set to false if isUsingConnection is false.
+            isUsingXAConnection = false;
+        }
+        this.isUsingXAConnection = isUsingXAConnection;
+        testName += "isUsingXAConnection=" + isUsingXAConnection + ";";
+        if (isUsingXAConnection)
+            setDataSource(ds);
+    }
+
+    /**
+     * Sets boolean property if Pooled Connection must be used to test bulk Copy
+     * 
+     * @param isUsingPooledConnection
+     *        <code>true</code> if Pooled connection object should be passed in BulkCopy constructor <code>false</code>
+     *        if otherwise
+     */
+    public void setUsingPooledConnection(boolean isUsingPooledConnection, ISQLServerDataSource ds) {
+        if (!isUsingConnection || isUsingXAConnection) {
+            // Set to false if isUsingConnection is false or isUsingXAConnection is true.

```

Comment:
this comment (and above in setUsingXAConnection) just says what the code does. Remove or explain why?